src/data/syncset/syncset.rs

```diff
@@ -0,0 +1,592 @@
+use crate::crypto::hash::hash;
+use super::Set;
+use super::Syncable;
+use super::errors::*;
+use super::node::Node;
+use super::path::*;
+
+pub struct SyncSet<Data: Syncable> {
+    root: Node<Data>,
+}
+
+// Round, the structure used to sync Syncsets
+#[derive(Debug, Clone)]
+pub struct Round<'a, 'b, Data: Syncable> {
+    pub view: Vec<Set<&'a Data>>,
+    pub add: Vec<&'b Data>,
+    pub remove: Vec<&'a Data>,
+}
+
+/// A Set based on Merkle trees, with efficient (O(K log N), K number of differences,
```

Comment:
This doc comment should go with the struct definition so that the generated rust doc is easier to read